### ISSUE TITLE
BUG: No protocol method DateTimeProtocol.plus- defined for type null

### ISSUE THREAD

**Issue by pedropombeiro** [status-im/status-react#4597]
[comment]: # (Please replace ... with your information. Remove < and >)

### User Story

As a user, I want to be able to use chat for long periods without running into unexpected behavior.

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Bug

[comment]: # (Describe the feature you would like, or briefly summarise the bug and what you did, what you expected to happen, and what actually happens. Sections below)
*Summary*: The app pops up the following error after prolonged usage:

![image](https://user-images.githubusercontent.com/138074/40871763-459b5654-6642-11e8-8961-6dc19d31cc51.png)

#### Expected behavior
[comment]: # (Describe what you expected to happen.)
The user is able to keep chatting without running into errors.

#### Actual behavior
[comment]: # (Describe what actually happened.)
@andytudhope has [mentioned](https://status-im.slack.com/archives/C90325XL6/p1527874458000593) that the behavior above happens after prolonged use.

### Reproduction
[comment]: # (Describe how we can replicate the bug step by step.)

TODO: Maybe @andytudhope has some insight here.

### Additional Information
[comment]: # (Please do your best to fill this out.)
* Status version: 0.9.19 (6027)
[comment]: # (Android or iOS.)
* Operating System: Android

**Comment by rachelhamlin** [comment]
I get this error frequently on an account that was created on the May 16 nightly. 

Have not seen it on my 0.9.19 account.

**Comment by lukaszfryc** [comment]
Adding it to Beta milestone as I can reproduce it every time when I recover account

**Comment by oskarth** [comment]
What's the state of this issue?

**Comment by mandrigin** [comment]
@oskarth I was planning to take a look at it today or tomorrow.

**Comment by janherich** [comment]
@oskarth I'm quite confident that I will finally solve it today.

**Comment by mandrigin** [comment]
@janherich awesome! just go for it! 🏅

**Comment by lukaszfryc** [comment]
https://github.com/status-im/status-react/issues/4565 is most probably a duplicate of this issue. Anyway, let's remember to take it into account while testing the fix.

**Comment by Serhy** [comment]
We reproduced this issue and as in #4565 mentioned - two issues represented on both screenshots ("screwed" message UI) and `DateTimeProtocl.Plus-` errors are connected to each other. 
Everything is connected to the simultaneous contact requests contact receives along with offline messages for the recovered accounts. It doesn't matter if messages sent/received in diferent networks.

There is the next scenario (it could be reproduced even on a single device - but 3 devices below provided for more clearer understanding of where the issue is):
1) Open the fresh status app (with no data) on Device1, Device2, Device3
2) Create Contact1, Contact2 and Contact3 respectively
3) **Contact2** sends a contact request to **Contact1** followed up with 10 text messages
4) **Contact3** sends a contact request to **Contact1** followed up with 10 text messages
5) Contact1 navigates to 1-1 chat with Contact2 and tap on 'Add to Contacts' 
6) Contact1 navigates to 1-1 chat with Contact3 and tap on 'Add to Contacts' 
7) Contact1 remembers a passphrase.
8) Contact1 re-installs the app (so no data is present in app build)
9) Contact1 recovers the account and waits on Home screen (Note: at this stage on the Contact1 side two 1-1 chats re-appears with Contact2 and Contact3 he received requests with messages before) 
10) Contact1 navigates to 1-1 with Contact2 **OR** Contact3

**Actual result:** Unknown 10 messages are present at the top of chat screen (https://user-images.githubusercontent.com/7532782/40782164-cdcaf4cc-64de-11e8-8d73-e6e92006feeb.png)

11) Contact2 sends a text message to Contact1

**Actual result:** `DateTimeProtocol.plus- defined for type null` error appears (https://user-images.githubusercontent.com/7532782/40782214-e6495642-64de-11e8-84e4-a5aadc174840.png)
Sending a text message again results in the same error and no message received by Contact1



<blockquote></blockquote>
<blockquote></blockquote>
